dace/sdfg/state.py

```diff
@@ -2587,14 +2595,25 @@ def inline(self) -> Tuple[bool, Any]:
             for b_edge in parent.in_edges(self):
                 parent.add_edge(b_edge.src, self.start_block, b_edge.data)
                 parent.remove_edge(b_edge)
-            # Redirect all edges exiting the region to instead exit the end state.
-            for a_edge in parent.out_edges(self):
-                parent.add_edge(end_state, a_edge.dst, a_edge.data)
-                parent.remove_edge(a_edge)
-
-            for node in to_connect:
-                parent.add_edge(node, end_state, dace.InterstateEdge())
-
+            
+            end_state = None
+            if len(to_connect) > 0:
+                end_state = parent.add_state(self.label + '_end')
+                # Redirect all edges exiting the region to instead exit the end state.
+                for a_edge in parent.out_edges(self):
+                    parent.add_edge(end_state, a_edge.dst, a_edge.data)
+                    parent.remove_edge(a_edge)
+
+                for node in to_connect:
+                    parent.add_edge(node, end_state, dace.InterstateEdge())
+            else:
+                dead_blocks = [succ for succ in parent.successors(self) if parent.in_degree(succ) == 1]
+                while dead_blocks:
+                    layer = list(dead_blocks)
+                    dead_blocks.clear()
+                    for u in layer:
+                        dead_blocks.extend([succ for succ in parent.successors(u) if parent.in_degree(succ) == 1])
+                        parent.remove_node(u)
```

Comment:
Should this not be part of a transformation / pass like dead state elimination? I am ok with this here now but I still would want a TODO note or TODO(later)

Reply:
Good point - I am placing a TODO and will adjust this in the PR that adapts the passes to these changes.